webpack.config.js

```diff
@@ -43,13 +38,58 @@ function generateWebpackConfigForCanister(name, info) {
       path: path.join(__dirname, "dist"),
     },
     devServer: {
+
+      // Set up a proxy that redirects API calls and /index.html to the
+      // replica; the rest we serve from here.
+      onBeforeSetupMiddleware: (devServer) => {
+          const dfxJson = './dfx.json';
+          let replicaHost;
+
+          try {
+              replicaHost = require(dfxJson).networks.local.bind;
+          } catch (e) {
+              throw Error(`Could get host from ${dfxJson}: ${e}`);
+          }
+
+          if(!replicaHost.startsWith("http://")) {
+              replicaHost = `http://${replicaHost}`;
+          }
+
+          const canisterIdsJson = './.dfx/local/canister_ids.json';
+
+          let canisterId;
+
+          try {
+              canisterId = require(canisterIdsJson).internet_identity.local;
+          } catch (e) {
+              throw Error(`Could get canister ID from ${canisterIdsJson}: ${e}`);
+          }
+
+          // basically everything _except_ for index.js, because we want live reload
+          devServer.app.get(['/', '/index.html', '/faq', '/faq', 'about' ], HttpProxyMiddlware.createProxyMiddleware( {
+              target: replicaHost,
+              pathRewrite: (pathAndParams, req) => {
+                  let queryParamsString = `?`;
+
+                  const [path, params] = pathAndParams.split("?");
+
+                  if (params) {
+                      queryParamsString += `${params}&`;
+                  }
+
+                  queryParamsString += `canisterId=${canisterId}`;
```

Comment:
This is so that we can access the dev server on an url without the parameter right?
But shouldn't there also be changes in the `selenium.test.ts` file? Otherwise we have the canisterID twice.

Reply:
Good point!

Reply:
Ah, actually no. Looks like the selenium tests directly connect to the replica. We can homogenize everything at some point (I'd like a setup like the [nns-dapp](https://github.com/dfinity/nns-dapp/tree/main/e2e-tests))